<launch>
	<rosparam command="load" file="$(find four_wheel_omni)/config/params.yaml" />
	<arg name="joy_dev" default="/dev/input/js2"/>
	<arg name="arduino" default="true"/>
	<arg name="joystick" default="true"/>
	<arg name="joy_control" default="true"/>
	<arg name="four_wheel_omni" default="true"/>
	<arg name="odom" default="true"/>

	<group if="$(arg joystick)">
		<node name="joystick" pkg="joy" type="joy_node" required="true">
			<param name="dev" value="$(arg joy_dev)"/>
		</node>
	</group>

	<group if="$(arg joy_control)">
		<node name="joy_twist_publisher" pkg="four_wheel_omni" type="joy_twist_publisher" output="screen" required="true"/>
	</group>

	<group if="$(arg four_wheel_omni)">
		<node name="four_wheel_omni" pkg="four_wheel_omni" type="four_wheel_omni" output="screen" required="true"/>
	</group>

	<group if="$(arg odom)">
		<node name="odometry" pkg="four_wheel_omni" type="odometry" output="screen" required="true"/>
	</group>

	<group if="$(arg arduino)">
		<node name="aruduino" pkg="rosserial_arduino" type="serial_node.py" required="true"  output="screen">
			<param name="port" value="/dev/ttyACM0"/>
			<param name="baud" value="80000000"/>
		</node>
	</group>
</launch>
